### DISCUSSION TITLE
How does Excalidraw re-render after cache revalidation?

### DISCUSSION THREAD

**Discussion by inwardmovement** [excalidraw/excalidraw#3893] [General]
It seems Excalidraw uses Workbox to manage cache revalidation, but then how is the app reloaded to display the new cache?

**Comment by dwelle** [comment]
Currently we're reloading the browser, which isn't ideal. If anyone wants knows their way around PWAs and want to improve this, that'd be great!

**Reply by inwardmovement** [reply]
According to [this answer](https://stackoverflow.com/questions/68663528/how-to-implement-the-cache-then-network-strategy-with-workbox-cli/#comment121387903_68684275) from the Workbox creator you could re-render with React instead of reloading.

Can you tell me where is the code taking care of reloading when cache revalidates?

**Reply by dwelle** [reply]
https://github.com/excalidraw/excalidraw/blob/8b3f236cd8ed2f41448eeac744ec16766869ab99/src/excalidraw-app/pwa.ts#L23-L25

> Once the new precached assets are available, based on the presence of the waiting service worker, you could re-run whatever client-side rendering logic you have and it will pick up the new assets (**assuming they have the same URL**).

I believe webpack generates new filenames every change so the emphasized part won't apply.
